core/src/com/unciv/ui/victoryscreen/VictoryScreen.kt

```diff
@@ -175,22 +177,89 @@ class VictoryScreen(val worldScreen: WorldScreen) : PickerScreen() {
+        val playerCiv = gameInfo.civilizations.filter { it.isCurrentPlayer() }
+
+        contentsTable.clear()
+
+        if (gameInfo.gameParameters.demographicsEnabled) contentsTable.add(buildDemographicsTable(majorCivs, playerCiv))
+        else contentsTable.add(buildRankingsTable(majorCivs))
+    }
+
+    private fun buildDemographicsTable(majorCivs: List<CivilizationInfo>, playerCiv: List<CivilizationInfo>): Table {
+        val rankLabels = arrayOf("Demographic", "Rank", "Value", "Best", "Average", "Worst")
+        val demographicsTable = Table().apply { defaults().pad(5f) }
+        for (n in rankLabels.indices)   {
+            if (n == 0) {
+                val demoLabel = Table().apply { defaults().pad(5f) }
+                demoLabel.add(rankLabels[n].toLabel()).row()
+                demoLabel.addSeparator().fillX()
+                demographicsTable.add(demoLabel)
+                for (category in RankingType.values()) {
+                    val headers = Table().apply { defaults().pad(5f) }
+                    headers.add(category.name.replace('_', ' ').toLabel()).row()
+                    headers.addSeparator().fillX()
+                    demographicsTable.add(headers)
+                    }
+            } else {
+                demographicsTable.row()
+                demographicsTable.add(rankLabels[n].toLabel())
+                for (category in RankingType.values()) {
+                    when (n) {
+                        1 -> { //finds the current player's rank in each category
+                            for ((rankCount, civ) in majorCivs.sortedByDescending { it.getStatForRanking(category) }.withIndex()) {
+                                if (civ.isCurrentPlayer()) {
+                                    val rank = (rankCount + 1).toLabel()
+                                    demographicsTable.add(rank)
+                                }
+                            }
+                        }
+                        2 -> { //finds the current player's value in each category
+                            for (civ in playerCiv) demographicsTable.add(getCivGroup(civ, ": " + civ.getStatForRanking(category).toString(), playerCivInfo)).fillX()
+                        }
+                        3 -> { //finds civ with the best value in each category
+                            for ((rankCount, civ) in majorCivs.sortedByDescending { it.getStatForRanking(category) }.withIndex()) {
+                                if (rankCount == majorCivs.indices.first) {
+                                    demographicsTable.add(getCivGroup(civ, ": " + civ.getStatForRanking(category).toString(), playerCivInfo)).fillX()
+                                }
+                            }
+                        }
+                        4 -> { //calculates the average value in each category
+                            var totalScore = 0
+                            for (civ in majorCivs) totalScore += civ.getStatForRanking(category)
```

Comment:
I'm pretty sure there's an average() function for lists, so majorCivs.map{it.getStatForRanking(category)}.avg() or something should do it

Reply:
Can't find such an average - sumOf / count would iterate twice

Reply:
Code simplified to one line with no loops.